x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/job/config/Job.java

```diff
@@ -136,6 +137,7 @@
         parser.declareString(Builder::setResultsIndexName, RESULTS_INDEX_NAME);
         parser.declareBoolean(Builder::setDeleting, DELETING);
         parser.declareBoolean(Builder::setAllowLazyOpen, ALLOW_LAZY_OPEN);
+        parser.declareStringOrNull(Builder::setBlockReason, BLOCK_REASON);
```

Comment:
This isn't required. When writing the XContent body, `block_reason` isn't written when its `null`. So, it should never be a `null` value when parsing from XContent

Reply:
When we update the job document to clear `block_reason` we do set `block_reason` to `null` explicitly.

Reply:
@dimitris-athanasiou true, but is `null` ever written to the xcontent builder and thus written to the doc? From what I can tell, it is not. It is simply not written to the doc at all. 

There is a big difference between `block_reason: null` and `block_reason` not being in the doc at all.

Reply:
It is. See `JobConfigProvider.updateJobBlockReason`.

Reply:
Ah, we are writing a map directly and updating the doc. 

Why are we explicitly saying `null` for the value instead of removing the field?

We now have two things that indicate that it isn't blocked, the field not existing or the field being explicitly null.